### ISSUE TITLE
Enhancement: Don't re-create DM rooms of they already exist

### ISSUE THREAD

**Issue by ctron** [8go/matrix-commander#140]
Before being able to send a DM to a user, a "DM room" most be created. This can be achieved using `--room-dm-create`.

However, calling that repeatedly will cause multiple rooms to be created. The user is spammed with "create new room" requests.

If would be great if there was an `--when-not-found`  (or other name) setting, which doesn't create a new DM room is one is already found. This should include rooms where the setup process is not completed yet, so the user is invited, but did not yet join.

**Comment by 8go** [comment]
This is not really doable, at least not right now.

Currently `mc` would have to call the equivalent to `--joined-rooms` to get a list of all rooms, then for each room call the equivalent to `--joined-members` and then see if the member are 2 in count and match the corresponding 2 users. Complicated and slow. 

Furthermore, in the first step that a DM is created, it creates a brand new room and sends an invite to the receiver. As long as the receiver does not accept the invite, there is no way to know for the `mc` if such an invite was sent. The receiver can query for outstanding invites. But `mc` cannot query for sent-and-open invites. 

Further complicating the situation, if the receiver rejects the invite, the room remains, with 1 user (the creator, i.e. `mc`) in it as member. 

But there might be some improvements coming in the future. See

- https://github.com/poljar/matrix-nio/issues/421
- https://github.com/poljar/matrix-nio/pull/451

This might help in the case of the receiver has accepted the invitation. I doubt that this will help in the case that a DM invitation has been sent but the receiver has not accepted the invitation. So, if one repeatedly sends `--room-dm-create` but the receiver ignores all the invites, the rooms will pile up and the invitations will pile up (you are spamming the receiver with invites and spamming the server with useless rooms). 

So in short, if the PR https://github.com/poljar/matrix-nio/pull/451 is pulled, `mc` can check if an established DM exists. And if so, avoid creating a new room and avoid sending an invitation. But it will not fix the issue when DM is open as unaccepted invitation. 

Also it is a bit confusing, if the user specifies 2 different `--name` or `--alias` for 2 different calls to `room-dm-create`. Does the user want 2 DMs to the same person, but with 2 different names?

**Comment by 8go** [comment]
See release 0.7.5

It avoids creating new DM rooms if one has been already established, invitation has been accepted.
It is slow because it has more server API queries.
